pkgs/build-support/fetchsource/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, pkgs, python3, nix-prefetch-scripts }:
+with pkgs;
+stdenv.mkDerivation {
+  name = "nix-prefetch-source";
+  buildInputs = [ makeWrapper ];
+  buildCommand = ''
+    mkdir -p $out/bin
+    makeWrapper ${./nix-prefetch-source} $out/bin/nix-prefetch-source \
+      --prefix PATH : ${python3}/bin \
+      --prefix PATH : ${nix-prefetch-scripts}/bin \
+    ;
+  '';
+  passthru = {
+    import = path:
+      let
+        json = lib.importJSON path;
+        fetchFunction = builtins.getAttr json.type pkgs;
```

Comment:
Yeah I can see how the generated file will probably come from an untrusted source in the future. Anyways, it's an improvement and shouldn't be a blocker for this PR.